tla-types/src/main/scala/at/forsyte/apalache/tla/typedbuilder/TT1OperatorSignatureMatcher.scala

```diff
@@ -0,0 +1,819 @@
+
+    /** Sets */
+    case TlaSetOper.in | TlaSetOper.notin                      => matchSetMembershipSig(args)
+    case TlaSetOper.subseteq                                   => matchSetCmpSig(args)
+    case TlaSetOper.cup | TlaSetOper.cap | TlaSetOper.setminus => matchBinarySetManipSig(args)
+    case TlaSetOper.enumSet                                    => matchSetEnumSig(args)
+    case TlaSetOper.filter                                     => matchSetFilterSig(args)
+    case TlaSetOper.map                                        => matchSetMapSig(args)
+    case TlaSetOper.powerset                                   => matchPowSetSig(args)
+    case TlaSetOper.union                                      => matchBigUnionSig(args)
+
+    /** Finite sets */
+    case TlaFiniteSetOper.isFiniteSet => matchSetPredicateSig(args)
+    case TlaFiniteSetOper.cardinality => matchCardinalitySig(args)
+
+    /** Functions */
+    case TlaSetOper.funSet => matchFunSetSig(args)
+    case TlaFunOper.funDef => matchFunDefSig(args)
+
+    /** Records */
+    case TlaFunOper.enum   => matchRecEnumSig(args)
+    case TlaSetOper.recSet => matchRecSetSig(args)
+
+    /** Tuples */
+    case TlaSetOper.times => matchCartesianProdSig(args)
+    // Tuple always makes a tuple, never a sequence. A special builder method exists to
+    // construct sequences and "cast" the tuple type to Seq
+    case TlaFunOper.tuple => matchTupleSig(args)
+
+    /** Control */
+    case TlaControlOper.ifThenElse    => matchIteSig(args)
+    case TlaControlOper.caseWithOther => matchCaseOtherSig(args)
+    case TlaControlOper.caseNoOther   => matchCaseNoOtherSig(args)
+
+    /** Oper */
+    case TlaOper.apply => matchOperAppSig(args)
+
+    /** Sequences */
+    case TlaSeqOper.concat    => matchConcatSig(args)
+    case TlaSeqOper.head      => matchHeadSig(args)
+    case TlaSeqOper.tail      => matchTailSig(args)
+    case TlaSeqOper.len       => matchLenSig(args)
+    case TlaSetOper.seqSet    => matchSeqSetSig(args)
+    case TlaSeqOper.append    => matchAppendSig(args)
+    case TlaSeqOper.subseq    => matchSubSeqSig(args)
+    case TlaSeqOper.selectseq => matchSelectSeqSig(args)
+
+    /** Overloaded */
+    case TlaFunOper.app    => matchApplicativeAppSig(args)
+    case TlaFunOper.except => matchExceptOverloadedSig(args)
+    case TlaFunOper.domain => matchApplicativeDomainSig(args)
+
+    /** Recursion */
+    case TlaFunOper.recFunDef => matchFunDefSig(args) // same signature as non-recursive function definitions
+    //    case TlaFunOper.recFunRef => resolved directly in builder via type-hint
+
+    // TODO
+    /** APALACHE */
+
+    // TODO
+    /** TLC */
+    //    case TlcOper.permutations => OperT(TupT(SetT(t)), SetT(FunT(t, t)))
+    //    case TlcOper.atat => OperT(TupT(FunT(t1, t2), FunT(t1, t2)), FunT(t1, t2))
+    //    case TlcOper.colonGreater => OperT(TupT(t1, t2), FunT(t1, t2))
+    //    case TlcOper.print => OperT(TupT(StrT, t), t)
+
```

Comment:
This would have clashed with proposed changes to `BmcOper`, where the class of operators got renamed and trimmed, so I expect this to be follow-up.

Reply:
If this is going to remain commented out when the PR is accepted, please add a few lines to indicate why it's here and when it can be removed/ implemented.